tests/models/test_taskinstance.py

```diff
@@ -855,6 +855,20 @@ def test_check_task_dependencies(self, trigger_rule, successes, skipped,
         self.assertEqual(completed, expect_completed)
         self.assertEqual(ti.state, expect_state)
 
+    def test_are_dependents_done(self):
+        dag = DAG(dag_id='test_dag')
+        upstream_task = DummyOperator(dag=dag, task_id='upstream_task', start_date=DEFAULT_DATE)
+        downstream_task = DummyOperator(dag=dag, task_id='downstream_task', start_date=DEFAULT_DATE)
+        upstream_task >> downstream_task
+        uti = TI(upstream_task, DEFAULT_DATE)
+        dti = TI(downstream_task, DEFAULT_DATE)
+        for state in State.task_states:
```

Comment:
This test is a bit hard to work out what it is doing, and should instead be refactored to use `@parameterize.expand`, something like this:

```python
    @parameterize.expand(
        (State.SUCCESS, True),
        (State.SKIPPED, True),
        (State.RUNNING, True),
     def test_are_dependents_done(self, current_state, expected_outcome):
```

Reply:
OK I can change this. Initially, I was thinking of using `@parameterize.expand` too, but saw that there are a lot of task states:
```
task_states = (
        SUCCESS,
        RUNNING,
        FAILED,
        UPSTREAM_FAILED,
        SKIPPED,
        UP_FOR_RETRY,
        UP_FOR_RESCHEDULE,
        QUEUED,
        NONE,
        SCHEDULED,
    )
``` which is why i decided to use a `for` loop that enumerates `State.task_states` for convenience.

Reply:
Do you think we need to test every state, or just a few?

Reply:
Just a few is sufficient actually.